### DISCUSSION TITLE
How to dynamically change output format on resized terminal

### DISCUSSION THREAD

**Discussion by bitfoo1** [gokcehan/lf#1551] [Q&A] [answered]
Hi,

I found the recol function in the [doc.md](https://github.com/gokcehan/lf/blob/master/doc.md?plain=1#L1402)
```comand
	cmd recol %{{
	    if [ $lf_width -le 80 ]; then
	        lf -remote "send $id set ratios 1:2"
	    elif [ $lf_width -le 160 ]; then
	        lf -remote "send $id set ratios 1:2:3"
	    else
	        lf -remote "send $id set ratios 1:2:3:5"
	    fi
	}}
```
But how can I send those command anytime the my terminal is getting resized? 
I tried to use `on-redraw` but that didn't change anything. 
Terminal is `wezterm` in version `20230712-072601-f4abf8fd`

Thanks!

**Comment by DusanLesan** [comment]
That's strange. I had almost the same code in my on-redraw working before, but now it doesn't work for me either.
I tried running lf with:
```
cmd on-cd &{{
	notify-send ">$lf_width<"
}}

cmd on-redraw &{{
	notify-send ">$lf_width<"
}}
```
However, the redraw was never called, even though it was working from on-cd.

**Comment by joelim-work** [comment, marked as answer]
The `on-redraw` hook was recently added in #1479 and is not yet released, see https://github.com/gokcehan/lf/issues/1427#issue-1899760013 for the upcoming changes.

For now you will have to build `lf` from source if you want to use `on-redraw`.

**Comment by bitfoo1** [comment]
Haha 🤣, I could have read the changelog. Sorry for taking you time on this. 
I guess, @DusanLesan has maybe another issue. But for me it's clear enought. 

Thank you! 

P.S: I read #1329. I would recommend to open a official chat channel e.g. on Matrix, Discord or Telegram etc., so that users do have easier access to others and could help. Especially with configuration/script errors. And general advice questions.

Again Tanks

**Reply by DusanLesan** [reply]
Oh I was using unreleased version and tested on-redraw and later forgot it is not yet out...

**Reply by bitfoo1** [reply]
A classic one. I also did have that on a different project.
